lib/cards/useGridMetrics.js

```diff
@@ -0,0 +1,87 @@
+import { throttle } from 'lodash';
+import { getCurrentInstance, inject, onBeforeUnmount, onMounted, provide, ref, watch } from 'vue';
+import useKResponsiveElement from '../composables/useKResponsiveElement';
+
+/**
+ * Composable to provide the context for calculating card metrics inside a grid.
+ * It allows cards to register and unregister their get metrics functions, and with
+ * this, we calculate the greatest metrics among all registered cards, so cards can then
+ * use this information to calculate their own metrics and have a consistent layout.
+ */
+export default function useGridMetrics() {
+  const registeredCards = ref({});
+
+  const registerCard = (uid, { getMetrics }) => {
+    registeredCards.value = {
+      ...registeredCards.value,
+      [uid]: {
+        getMetrics,
+      },
+    };
+  };
+  const unregisterCard = uid => {
+    const newRegisteredCards = { ...registeredCards.value };
+    delete newRegisteredCards[uid];
+    registeredCards.value = newRegisteredCards;
+  };
+
+  const selectionControlWidth = ref(0);
+
+  const calculateGridMetrics = () => {
+    let newSelectionControlWidth = 0;
+    const cards = Object.values(registeredCards.value);
+    for (const card of cards) {
+      const metrics = card.getMetrics();
+      if (metrics.selectionControlWidth > newSelectionControlWidth) {
+        newSelectionControlWidth = metrics.selectionControlWidth;
```

Comment:
I think you could add a `break;` here if I'm understanding correctly that, once `newSelectionControlWidth` is set, then we know we need to set the control-area width on all of the cards.

Reply:
Oh a break is not needed here, because we are finding the greater width so we need to go through all cards.